services/src/main/java/org/exoplatform/webconferencing/WebConferencingService.java

```diff
@@ -0,0 +1,1235 @@
+    }
+    json.put("participants", jsonParts);
+
+    return json;
+  }
+
+  /**
+   * Json to call.
+   *
+   * @param json the json
+   * @return the call info
+   * @throws Exception the exception
+   */
+  protected CallInfo jsonToCall(JSONObject json) throws Exception {
+    String id = json.getString("id");
+    String title = json.getString("title");
+    String avatarLink = json.getString("avatarLink");
+    String ownerType = json.getString("ownerType");
+    String ownerLink = json.getString("ownerLink");
+    String providerType = json.getString("providerType");
+    String state = json.optString("state", null);
+
+    // Owner
+    JSONObject jsonOwner = json.getJSONObject("owner");
+    String ownerId = jsonOwner.getString("id");
+    IdentityInfo owner;
+    if (OWNER_TYPE_CHATROOM.equals(ownerType)) {
+      // XXX room/space members not interesting in this context
+      owner = roomInfo(ownerId, jsonOwner.getString("name"), jsonOwner.getString("title"), new String[0], id);
+    } else if (OWNER_TYPE_SPACE.equals(ownerType)) {
+      owner = getSpaceInfo(ownerId);
+    } else if (UserInfo.TYPE_NAME.equals(ownerType)) {
+      owner = getUserInfo(ownerId);
+    } else {
+      LOG.error("Unexpected call owner, type: " + ownerType + ", id: " + ownerId);
+      throw new CallInfoException("Unexpected call owner: " + ownerId);
+    }
+
+    CallInfo call = new CallInfo(id, title, owner, ownerType, ownerLink, avatarLink, providerType);
+    if (state != null) {
+      call.setState(state);
+    }
+    // Participants
+    JSONArray jsonParts = json.getJSONArray("participants");
+    for (int i = 0; i < jsonParts.length(); i++) {
+      JSONObject jsonPart = jsonParts.getJSONObject(i);
+      String partId = jsonPart.getString("id");
+      String partType = jsonPart.getString("type");
+      if (UserInfo.TYPE_NAME.equals(partType)) {
+        UserInfo user = getUserInfo(partId);
+        if (user != null) {
+          String partState = jsonPart.optString("state", null);
+          if (partState != null) {
+            user.setState(partState);
+          }
+          call.addParticipant(user);
+        } else {
+          LOG.warn("Cannot add call participant as eXo user: " + partId);
+          call.addParticipant(new ParticipantInfo(providerType, partId));
+        }
+      } else {
+        call.addParticipant(new ParticipantInfo(providerType, partId));
+      }
+    }
+
+    return call;
+  }
+
+  /**
+   * Save call.
+   *
+   * @param call the call
+   * @throws Exception the exception
+   */
+  protected void saveCall(CallInfo call) throws Exception {
+    saveOwnerCallId(call.getOwner().getId(), call.getId());
+    final String initialGlobalId = Scope.GLOBAL.getId();
+    try {
+      JSONObject json = callToJSON(call);
+      String safeCallId = URLEncoder.encode(call.getId(), "UTF-8");
+      settingService.set(Context.GLOBAL, Scope.GLOBAL.id(CALL_ID_SCOPE_NAME), safeCallId, SettingValue.create(json.toString()));
```

Comment:
Why do we use Settings to save calls ? It is not the purpose of the settings. Calls should probably have their dedicated tables.